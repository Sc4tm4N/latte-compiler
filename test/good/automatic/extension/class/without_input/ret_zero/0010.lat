class Node {
  int a;

  void printMe() {
    printInt(a);
  }

  void setMe1(int b) {
    self.a = b;
  }

  void setMe2(int a) {
    a++;
    self.a = a;
  }
}

int main() {
  Node node = new Node;
  printInt(node.a);
  node.a = 10;
  printInt(node.a);
  node.setMe1(100);
  printInt(node.a);
  node.setMe2(1000);
  printInt(node.a);
  return 0;
}
