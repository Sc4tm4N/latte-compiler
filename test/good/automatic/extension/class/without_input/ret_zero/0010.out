0
10
100
1001
